rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/jboss/ResolveJBossLegacyEjbXmlRuleProvider.java

```diff
@@ -136,10 +136,11 @@ public void perform(GraphRewrite event, EvaluationContext context, XmlFileModel
             if (StringUtils.isNotBlank(ejbName))
             {
                 LOG.info("Looking up name: " + ejbName);
-                for (EjbSessionBeanModel ejb : ejbSessionBeanService.findAllByProperty(EjbMessageDrivenModel.EJB_BEAN_NAME, ejbName))
+                for (EjbSessionBeanModel ejb : ejbSessionBeanService.findAllByProperty(EjbSessionBeanModel.EJB_BEAN_NAME, ejbName))
                 {
                     String jndi = $(ejbRef).child("jndi-name").content();
                     String localJNDI = $(ejbRef).child("local-jndi-name").content();
+                    String remoteBindingJNDI = $(ejbRef).child("remote-binding").child("jndi-name").content();
```

Comment:
I was looking too long but this is just another type of jndi to use as global one.

Reply:
Ah, good find.